web/server/vue-cli/src/components/Report/ReportFilter/Filters/BaselineTagItems.vue

```diff
@@ -1,83 +1,167 @@
+    runId: { type: Number, required: true },
     selectedItems: { type: Array, default: null },
-    selectedRunItems: { type: Array, default: null },
-    bus: { type: Object, required: true },
-    loadEventBus: { type: Object, required: true },
-    fetchTags: { type: Function, required: true }
+    limit: { type: Number, required: true }
   },
   data() {
     return {
       loading: false,
       search: {
-        placeHolder : "Search for run tags...",
+        placeHolder : "Search by tag name...",
         filterItems: this.filterItems
       },
       tags: [],
-      reloadItems: false,
+      dateFilter: null,
       filterOpt: {}
     };
   },
   computed: {
-    selectedRunIds() {
-      return this.selectedRunItems.map(i => i.runIds).flat();
+    ...mapState({
+      reportFilter(state) {
+        return state[this.namespace].reportFilter;
+      }
+    })
+  },
+  watch: {
+    async runId() {
+      this.tags = await this.fetchTags(this.filterOpt);
+    },
+
+    async dateFilter() {
+      this.tags = await this.fetchTags(this.filterOpt);
     }
   },
 
   async mounted() {
-    this.bus.$on("update", () => this.reloadItems = true);
+    this.tags = await this.fetchTags(this.filterOpt);
+  },
 
-    // Initalize tags.
-    this.tags = await this.fetchTags();
+  methods: {
+    async getRunTagFilter() {
+      const query = this.filterOpt.query;
 
-    // Register on load events to update the items.
-    this.loadEventBus.$on("load", async () => {
-      if (!this.reloadItems) return;
+      if (!query && !this.dateFilter)
+        return null;
 
-      this.reload();
-      this.reloadItems = false;
-    });
+      let stored = null;
+      if (this.dateFilter) {
+        const date = parse(this.dateFilter, "yyyy-MM-dd", new Date());
+        const midnight = endOfDay(date);
 
-    this.loadEventBus.$on("reload", async () => {
-      this.reload();
-      this.reloadItems = false;
-    });
-  },
+        stored = new DateInterval({
+          before: this.getUnixTime(midnight)
+        });
+      }
 
-  methods: {
-    filterItems(value) {
-      return this.fetchTags({ query: value ? [ `${value}*` ] : null });
+      const tags =
+        await ccService.getTags([ this.runId ], null, query, stored);
+
+      return tags.map(t => t.id.toNumber());
     },
 
-    async reload() {
-      this.tags = await this.fetchTags(this.filterOpt);
+    async fetchTags(opt={}) {
+      this.loading = true;
+      this.filterOpt = opt;
+
+      const reportFilter = new ReportFilter(this.reportFilter);
+      const limit = opt.limit || this.limit;
+      const offset = 0;
+
+      reportFilter.runTag = await this.getRunTagFilter();
+
+      return new Promise(resolve => {
+        ccService.getClient().getRunHistoryTagCounts([ this.runId ],
+          reportFilter, null, limit, offset, handleThriftError(res => {
+            resolve(res.map(tag => {
+              const id = tag.id.toNumber();
+              const name = tag.name || "N/A";
```

Comment:
What do you think about renaming the `N/A` to something like `Storage without a named tag`
If I have multiple store events without a tag I just see a list of `N/A`.